queries.sql

```diff
@@ -58,4 +58,127 @@ FROM
     animals
 WHERE
     (weigth_kg >= 10.4)
-    AND (weigth_kg <= 17.3);
\ No newline at end of file
+    AND (weigth_kg <= 17.3);
+
+begin;
+
+UPDATE
+    animals
+SET
+    species = 'unspecified';
+
+select
+    *
+from
+    animals;
+
+rollback;
+
+select
+    *
+from
+    animals;
+
+begin;
+
+UPDATE
+    animals
+SET
+    species = 'digimon'
+WHERE
+    name LIKE '%mon';
+
+select
+    *
+from
+    animals;
+
+UPDATE
+    animals
+SET
+    species = 'pokemon'
+WHERE
+    species IS NULL
+    OR species = '';
+
+select
+    *
+from
+    animals;
+
+commit;
+
+select
+    *
+from
+    animals;
+
+SELECT
+    COUNT(*)
+FROM
+    animals;
+
+SELECT
+    COUNT(*)
+FROM
+    animals
+WHERE
+    (escape_attempts = 0);
+
+SELECT
+    AVG(weigth_kg)
+from
+    animals;
+
+SELECT
+    SUM(escape_attempts)
+FROM
+    animals
+WHERE
+    (neutered = true);
+
+SELECT
+    SUM(escape_attempts)
+FROM
+    animals
+WHERE
+    (neutered = false);
+
+SELECT
+    MIN(weigth)
+FROM
+    animals
+WHERE
+    (neutered = true);
+
+SELECT
+    MAX(weigth_kg)
+FROM
+    animals
+WHERE
+    (neutered = true);
+
+SELECT
+    MAX(weigth_kg)
+FROM
+    animals
+WHERE
+    (neutered = FALSE);
+
+SELECT
+    MIN(weigth_kg)
+FROM
+    animals
+WHERE
+    (neutered = FALSE);
+
```

Comment:
- [ ] Same for the question **What is the minimum and maximum weight of each type of animal?**, It would be nice to use the `GROUP BY` keyword 
HIT: Your code could look like this :point_down: 
```sql
   SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals
   GROUP BY species;
```